tripy/tripy/frontend/tensor.py

```diff
@@ -265,8 +269,15 @@ def __repr__(self) -> str:
 
         data_list = self.tolist()
 
-        assert isinstance(self.trace_tensor.producer, Storage)
-        data_shape = self.trace_tensor.producer.shape
+        if isinstance(self.trace_tensor.producer, Storage):
```

Comment:
Does the memref have shape information? If so, it would probably be cleaner to use the memref directly rather than `tolist()`.